lib/common/session_model.dart

```diff
@@ -331,16 +331,22 @@ class SessionModel extends Model {
     }).then((value) => value as String);
   }
 
-  Future<String> approveDevice(String code) {
-    return methodChannel.invokeMethod('approveDevice', <String, dynamic>{
-      'code': code,
-    }).then((value) => value as String);
+  Future<String> approveDevice(String code) async {
+    if (isMobile()) {
+      return methodChannel.invokeMethod('approveDevice', <String, dynamic>{
+        'code': code,
+      }).then((value) => value as String);
+    }
+    return await ffiApproveDevice(code.toNativeUtf8());
```

Comment:
@atavism Can we use compute here to avoid UI freeze?

Reply:
Done.. thanks!